src/wp-includes/media.php

```diff
@@ -1958,6 +1971,104 @@ function wp_img_tag_add_loading_attr( $image, $context ) {
 	return $image;
 }
 
+/**
+ * Adds optimization attributes to an `img` HTML tag.
+ *
+ * @since 6.3.0
+ *
+ * @param string $image   The HTML `img` tag where the attribute should be added.
+ * @param string $context Additional context to pass to the filters.
+ *
+ * @return string Converted `img` tag with optimization attributes added.
+ */
+function wp_img_tag_add_loading_optimization_attrs( $image, $context ) {
+	$width              = preg_match( '/ width="([0-9]+)"/', $image, $match_width ) ? (int) $match_width[1] : '';
+	$height             = preg_match( '/ height="([0-9]+)"/', $image, $match_height ) ? (int) $match_height[1] : '';
+	$optimization_attrs = wp_get_loading_optimization_attributes(
+		'img',
+		array(
+			'width'  => $width,
+			'height' => $height,
+		),
+		$context
+	);
+
+	/**
+	 * Filters the loading optimization attributes to be added to an image.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string|bool $optimization_attrs   Attribute and it's values obtained from `wp_get_loading_optimization_attributes` function.
+	 * @param string      $image   The HTML `img` tag to be filtered.
+	 * @param string      $context Additional context about how the function was called or where the img tag is.
+	 */
+	$optimization_attrs = apply_filters( 'wp_img_tag_add_loading_optimization_attrs', $optimization_attrs, $image, $context );
+
+	/**
+	 * Filters the `loading` attribute value to add to an image. Default `lazy`.
+	 * This filter is added in for backward compatibility.
+	 *
+	 * Returning `false` or an empty string will not add the attribute.
+	 * Returning `true` will add the default value.
+	 * `true` and `false` usage supported for backward compatibility.
+	 *
+	 * @param string|bool $optimization_attrs['loading'] Current value for `loading` attribute for the image.
+	 * @param string      $image   The HTML `img` tag to be filtered.
+	 * @param string      $context Additional context about how the function was called or where the img tag is.
+	 */
+	$optimization_attrs['loading'] = apply_filters(
+		'wp_img_tag_add_loading_attr',
+		isset( $optimization_attrs['loading'] ) ? $optimization_attrs['loading'] : false,
+		$image,
+		$context
+	);
+
+	/**
+	 * Validate the values after filtering.
+	 */
+	if ( isset( $optimization_attrs['loading'] ) && $optimization_attrs['loading'] ) {
+		if ( ! in_array( $optimization_attrs['loading'], array( 'lazy', 'eager' ), true ) ) {
+			$optimization_attrs['loading'] = 'lazy';
+		}
+	}
```

Comment:
See above, rather than changing the values to something arbitrary, let's fall back to the default values determined by core if the filter output is invalid.

Based on my above comment and putting the filter output in a separate variable, you can simplify this: If the filter output is valid, overwrite what is in `$optimization_attrs` - otherwise, just keep using the original value.
```suggestion
	if ( in_array( $filtered_loading_attr, array( 'lazy', 'eager' ), true ) ) {
		$optimization_attrs['loading'] = $filtered_loading_attr,;
	}
```